### DISCUSSION TITLE
Unit Testing

### DISCUSSION THREAD

**Discussion by marchbold** [airsdk/Adobe-Runtime-Support#885] [Q&A]
Does anyone use any unit testing tools with AIR applications? 

Just wondering where this functionality sits currently as I'm looking to implement some tests on a new public project and don't know what the current best tools are. 

We have an internal tool that we use to test things but it is highly customised and not suited for this new project.

**Comment by leossmith** [comment]
In the very distant past, we used to work with [FlexUnit](https://flex.apache.org/flexunit/tutorial/flexunit/Unit-1.html). I am not sure if this applies to AIR or even works any more.

Others I know of but never used, are [asunit.org](http://asunit.org/) and [fluint](https://code.google.com/archive/p/fluint/)

**Reply by marchbold** [reply]
Ah wasn't aware of asunit, will have a look there, see if we can revive it.
